google-cloud-examples/src/main/java/com/google/cloud/examples/bigtable/HelloWorld.java

```diff
@@ -0,0 +1,172 @@
+import com.google.cloud.bigtable.admin.v2.models.Table;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.InstanceName;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowCell;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+
+public class HelloWorld {
+
+  public static void main(String... args) throws Exception {
+
+    final String GCLOUD_PROJECT_ID = args[0];
+    final String INSTANCE_ID = args[1];
+    final String TABLE_ID = "Hello-Bigtable";
+    final String COLUMN_FAMILY_ID = "cf1";
+    final String COLUMN_QUALIFIER = "greeting";
+    final String ROW_KEY_PREFIX = "rowKey";
+
+    if (args.length < 2) {
+      System.out.println("Missing required project id or instance id");
+      return;
+    }
+
+    // [START connecting_to_bigtable]
+    // Create the settings to configure a bigtable data client
+    BigtableDataSettings settings = BigtableDataSettings.newBuilder()
+        .setInstanceName(InstanceName.of(GCLOUD_PROJECT_ID, INSTANCE_ID)).build();
+
+    // Create bigtable data client
+    BigtableDataClient dataClient = BigtableDataClient.create(settings);
+
+    // Create the settings to configure a bigtable admin client
+    BigtableTableAdminSettings adminSettings = BigtableTableAdminSettings.newBuilder()
+        .setInstanceName(
+            com.google.bigtable.admin.v2.InstanceName.of(GCLOUD_PROJECT_ID, INSTANCE_ID)).build();
+
+    // Create bigtable admin client
+    BigtableTableAdminClient adminClient = BigtableTableAdminClient.create(adminSettings);
+    // [END connecting_to_bigtable]
+
+    try {
+      // Check if table exists, create table if does not exist
+      createTable(adminClient, TABLE_ID, COLUMN_FAMILY_ID);
+
+      // Write to table
+      writeToTable(dataClient, TABLE_ID, ROW_KEY_PREFIX, COLUMN_FAMILY_ID, COLUMN_QUALIFIER);
+
+      // Read single row from the table
+      readSingleRow(dataClient, TABLE_ID, ROW_KEY_PREFIX);
+
+      // Read entire table
+      readTable(dataClient, TABLE_ID);
+
+      // Delete table
+      deleteTable(adminClient, TABLE_ID);
+    } catch (Exception e) {
+      System.out.println("Exception while running HelloWorld: " + e.getMessage());
+    } finally {
+      dataClient.close();
+      adminClient.close();
+    }
+  }
+
+  public static Table createTable(BigtableTableAdminClient adminClient, String TABLE_ID,
+      String COLUMN_FAMILY_ID) {
+    // [START creating_a_table]
+    Table table = null;
+    if (!adminClient.exists(TABLE_ID)) {
+      CreateTableRequest createTableRequest =
+          CreateTableRequest.of(TABLE_ID).addFamily(COLUMN_FAMILY_ID);
+      System.out.println("Creating table: " + TABLE_ID);
+      table = adminClient.createTable(createTableRequest);
+    }
+    return table;
```

Comment:
Why do you need table to return here its not getting used and also if table exists it will return null. I think you should just make sure that table gets created. Also if table exists dont we need to delete and recreate it? what if its created with some other column family and contains some other unexpected rows?